### PR TITLE
[#10] Add missing Apache License | [#32] Enhance the build scripts

### PR DESCRIPTION
**Pull request by Dee-Pac**
Make sure you have checked all steps below.

### Checklist:

- [X] This pull request updates the documentation
- [ ] This pull request changes library dependencies
- [X] Title of the PR is of format (example) : [#25][Github] Add Pull Request Template
- [X] This PR includes the URL to the GitHub Issue 

### What is the purpose of this pull request?

- Ideally we'd like to have a separate commit per issue & also PR per issue. However, in this case 2 issues are affecting same set of files & one of the two tickets is a documentation change. Hence, I'm combining these issues in 1 PR to avoid overhead.
- [#10] Adds Apache License text to build/* scripts that was missing.
- [#32] Enhance the build scripts
- [#32] update ./docs/getting-started.md to have the build scripts instead of `mvn` commands.
- Please ignore the CI failure. It will be addressed in the next PR.
At this point - integration is done with Travis CI, but the yml file is missing - which I will commit in upcoming PR.

### How was this change validated?

- [#32] was validated via build script commands
```bash
build/gimel   #builds default configuration
build/gimel install #builds entire project
build/gimel install -pl gimel-logging #builds logging module
build/gimel install -pl gimel-dataapi/gimel-tools -am #builds all dependent modules + gimel-tools.
```

### Commit Guidelines
- [X] My commits all reference GH issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by Dee-Pac** [comment]
@rampallydheeraj 
Please ignore the CI failure. It will be addressed in the next PR.
At this point - integration is done with Travis CI, but the yml file is missing - which I will commit in upcoming PR.

**Comment by r39132** [comment]
@Dee-Pac do you have the travis.yml file in another commit?

File: build/install_dependencies [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#----------------------------function will check for error code & exit if failure, else proceed further----------------------------#
+
+#usage : Run immediately after executing any shell command - with 2 arguments.
+#Example: Check_error < pass $? from the shell command > < Custom Message for errorcode -gt 0 >
+
+check_error()
```
Comment (rampallydheeraj) [inline build/install_dependencies]: @Dee-Pac 

Nit: You can source the check_error() and write_log() functions by writing them in a different shell script, thereby those would be reusable

File: docs/getting-started.md [hunk 2]
```diff
@@ -49,7 +53,7 @@ Run below command to build
       <groupId>com.paypal.gimel</groupId>
       <artifactId>gimel-tools</artifactId> <!--Refer one of the below listed 3 versions, depending on the required spark version -->
       <version>1.2.0</version> <!--provides spark 2.2.0 compiled code-->
```
Comment (rampallydheeraj) [inline docs/getting-started.md]: Should the version go in here ?

**Comment by Dee-Pac** [comment]
@r39132 The travis file is in PR https://github.com/paypal/gimel/pull/37
@rampallydheeraj Merged the other PR before this is merged. So I've rebased this PR. 
I can see travis building now...

File: docs/getting-started.md (hunk 2 above)
Comment (Dee-Pac) [reply docs/getting-started.md]: Yes, this should.

**Comment by Dee-Pac** [comment]
@rampallydheeraj Please note the Travis CI is now successful.
Please note that `.travis.yml` is only building till module `gimel-dataapi/gimel-tools`. 
After this stage : there is no space on the container & build errors out. We need to fix that and ensure `all` modules in gimel build at some point.

**Pull request merged**